sqlglot/dialects/clickhouse.py

```diff
@@ -849,3 +876,7 @@ def partition_sql(self, expression: exp.Partition) -> str:
 
         def partitionid_sql(self, expression: exp.PartitionId) -> str:
             return f"ID {self.sql(expression.this)}"
+
+        def replacepartition_sql(self, expression: exp.ReplacePartition) -> str:
+            source = expression.args.get("source")
+            return f"REPLACE {self.sql(expression.expression)} FROM {source}"
```

Comment:
Keep in mind to avoid interpolating expressions, because that silently invokes `.sql()` without a dialect and may produce incorrect SQL.
```suggestion
            return f"REPLACE {self.sql(expression.expression)} FROM {self.sql(expression, 'source')}"
}"
```

Reply:
Will keep that in mind, thanks!